database/src/main/java/database/service/SqlDatabaseService.java

```diff
@@ -0,0 +1,438 @@
+            statement.executeUpdate(dropTableSQL);
+            return true;
+        } catch (SQLException e) {
+            LOG.error(UNABLE_DELETE_TABLE + ": {}", tableName);
+            throw new DeletionDatabaseException(UNABLE_DELETE_TABLE);
+        }
+    }
+
+    @Override
+    public <T extends BaseEntity> T addNewRecordToTable(T entity) {
+        if (entity.getId() != null) {
+            throw new IdProvidedManuallyException(ID_PROVIDED_MANUALLY);
+        }
+
+        Class<? extends BaseEntity> entityClass = entity.getClass();
+        String tableName = entityClass.getSimpleName();
+
+        String checkTableSQL = "SELECT COUNT(*) FROM information_schema.tables " +
+                "WHERE table_schema = 'entities' AND table_name = '" + tableName + "'";
+
+        String checkEmptyTableSQL = "SELECT COUNT(*) FROM " + tableName;
+
+        try (Connection connection = DriverManager.getConnection(dbURL, dbUserName, dbPassword);
+             Statement statement = connection.createStatement();
+             ResultSet resultSet = statement.executeQuery(checkTableSQL)) {
+
+            if (resultSet.next() && resultSet.getInt(1) > 0) {
+                try (ResultSet emptyTableResultSet = statement.executeQuery(checkEmptyTableSQL)) {
+                    int rowCount = 0;
+                    if (emptyTableResultSet.next()) {
+                        rowCount = emptyTableResultSet.getInt(1);
+                    }
+
+                    if (rowCount == 0) {
+                        String resetAutoIncrementSQL = "ALTER TABLE " + tableName + " AUTO_INCREMENT = 1";
+                        statement.executeUpdate(resetAutoIncrementSQL);
+                    }
+                }
+
+                String insertSQL = generateInsertSQL(entity, tableName);
+                int affectedRows = statement.executeUpdate(insertSQL, Statement.RETURN_GENERATED_KEYS);
+                if (affectedRows == 0) {
+                    throw new SQLException("Creating record failed, no rows affected.");
+                }
+
+                try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                    if (generatedKeys.next()) {
+                        entity.setId(generatedKeys.getInt(1)); // Получаем сгенерированный ID
+                    } else {
+                        throw new SQLException("Creating record failed, no ID obtained.");
+                    }
+                }
+            } else {
+                throw new DatabaseDoesNotExistException(TABLE_NOT_EXIST);
+            }
+        } catch (SQLException e) {
+            LOG.error(UNABLE_ADD_NEW_RECORD + ": {}", tableName, e);
+            throw new DatabaseOperationException(UNABLE_ADD_NEW_RECORD);
+        }
+
+        return entity;
+    }
+
+    //TODO: If entity id doesn't match parameter id, I don't get exception, why? The same problem in
+    // removeRecordFromTable method, when I use wrong id
```

Comment:
Check string 89 in Controller: `ctx.json(databaseService.updateRecordInTable(entity, entity.getId()));`
You ignore `id` from `path` and use `id` from `entity`. I think we can move validation into controller. Please do.